wallet-web/components/bond/utils.ts

```diff
@@ -0,0 +1,96 @@
+import { printableBalanceToNative } from '@nymproject/nym-validator-client/dist/currency'
+import { Gateway, MixNode } from '@nymproject/nym-validator-client/dist/types'
+import bs58 from 'bs58'
+import semver from 'semver'
+import { basicRawCoinValueValidation } from '../../common/helpers'
+import { NodeType } from '../../common/node'
+import { BondingInformation } from './NodeBond'
+
+export const validateAmount = (rawValue: string, minimum: string): boolean => {
+  // tests basic coin value requirements, like no more than 6 decimal places, value lower than total supply, etc
+  if (!basicRawCoinValueValidation(rawValue)) {
+    return false
+  }
+
+  // this conversion seems really iffy but I'm not sure how to better approach it
+  let nativeValueString = printableBalanceToNative(rawValue)
+  let nativeValue = parseInt(nativeValueString)
+
+  return nativeValue >= parseInt(minimum)
+}
+
+export const validateKey = (key: string): boolean => {
+  // it must be a valid base58 key
+  try {
+    const bytes = bs58.decode(key)
+    // of length 32
+    return bytes.length === 32
+  } catch {
+    return false
+  }
+}
+
+export const isValidHostname = (value: string) => {
+  // regex for ipv4 and ipv6 - source https://www.regextester.com/104038
+  const hostnameRegex =
```

Comment:
This is a personal bias, but I love this quote about regexes:
```
Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.
```
If we're doing proper ip validation, surely there exists some package for that? Also in our case, `foo.com` is a valid hostname : )

Reply:
:laughing: I like that expression too. I have heard it before

I try to avoid regex as much as possible! But I think in this case it might be a good option. Saying that I will look into some packages that we might be able to use as an alternative.

I have  updated this part so that it will now validate ips and hostnames.

I'm adding `foo.com` to the tests :D